Service/InstallationService.php

```diff
@@ -354,4 +632,134 @@ public function handleInstaller($file)
 
         return $installationService->install();
     }
+
+    /**
+     * Handles forced id's on object entities.
+     *
+     * @param ObjectEntity $objectEntity
+     *
+     * @return ObjectEntity
+     */
+    private function saveOnFixedId(ObjectEntity $objectEntity, array $hydrate = []): ObjectEntity
+    {
+        // This savetey dosn't make sense but we need it
+        if (!$objectEntity->getEntity()) {
+            $this->logger->error('Object can\'t be persisted due to missing schema');
+            $this->io->writeln(['', 'Object can\'t be persisted due to missing schema']);
+
+            return $objectEntity;
+        }
+
+        // Save the values
+        //$values = $objectEntity->getObjectValues()->toArray();
+        //$objectEntity->clearAllValues();
+
+        // We have an object entity with a fixed id that isn't in the database, so we need to act
+        if (isset($hydrate['id']) && !$this->em->contains($objectEntity)) {
+            $this->io->writeln(['Creating new object ('.$objectEntity->getEntity()->getName().') on a fixed id ('.$hydrate['id'].')']);
+
+            // save the id
+            $id = $hydrate['id'];
+            // Create the entity
+            $this->em->persist($objectEntity);
+            $this->em->flush();
+            $this->em->refresh($objectEntity);
+            // Reset the id
+            $objectEntity->setId($id);
+            $this->em->persist($objectEntity);
+            $this->em->flush();
+            $objectEntity = $this->em->getRepository('App:ObjectEntity')->findOneBy(['id' => $id]);
+
+            $this->io->writeln(['Defintive object id ('.$objectEntity->getId().')']);
+        } else {
+            $this->io->writeln(['Creating new object ('.$objectEntity->getEntity()->getName().') on a generated id']);
+        }
+
+        // We already dit this so lets skip it
+        unset($hydrate['_id']);
+
+        foreach ($hydrate as $key => $value) {
```

Comment:
This code is very, very ugly.
We should really clean this up

Reply:
Agreed

Reply:
I will write a Jira issue for cleaning this code